File: ROS/ros2_ws/src/my_cpp_pkg/src/AddTwoIntsClient.cpp
#include "example_interfaces/srv/add_two_ints.hpp"
#include "rclcpp/rclcpp.hpp"

using namespace example_interfaces::srv;

class AddTwoIntsClientNode : public rclcpp::Node
{
  public:

    AddTwoIntsClientNode()
    : Node("add_two_ints_client")
    {
      m_threads.push_back(std::thread(std::bind(&AddTwoIntsClientNode::callAddTwoIntsService, this, 2, 3)));
      m_threads.push_back(std::thread(std::bind(&AddTwoIntsClientNode::callAddTwoIntsService, this, 4, 5)));
    }

    void
    callAddTwoIntsService(const int a, const int b)
    {
      auto client = this->create_client<AddTwoInts>("add_two_ints");

      while (!client->wait_for_service(std::chrono::seconds(1)))
        RCLCPP_WARN(this->get_logger(), "Waiting for service add_two_ints.");

      auto request = std::make_shared<AddTwoInts::Request>();
      request->a   = a;
      request->b   = b;

      auto future = client->async_send_request(request);

      try
      {
        auto response = future.get();
        RCLCPP_INFO(this->get_logger(), "%d + %d = %ld", a, b, response->sum);
      }
      catch (const std::exception& error)
      {
        RCLCPP_ERROR(this->get_logger(), "Error calling the service");
      }
    }

  private:

    std::vector<std::thread> m_threads;
};

int
main(int argc, char** argv)
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<AddTwoIntsClientNode>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}

File: ros2_ws/src/my_py_pkg/my_py_pkg/add_two_ints_client.py
#!/usr/bin/env python3

import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts
from functools import partial
 
 
class AddTwoIntsClient(Node):
    def __init__(self):
        super().__init__("add_two_ints_client")
        self.call_add_two_ints_server(4, 6)

    def call_add_two_ints_server(self, a, b):
        client = self.create_client(AddTwoInts, "add_two_ints")

        while not client.wait_for_service(1.0):
            self.get_logger().warn("Waiting for server add_two_ints")

        request = AddTwoInts.Request()
        request.a = a
        request.b = b

        future = client.call_async(request)
        future.add_done_callback(partial(self.callback_call_done, a=a, b=b))

    def callback_call_done(self, future, a, b):
        try:
            response = future.result()
            self.get_logger().info(str(a) + " + " + str(b) + " = " + str(response.sum))
        except Exception as e:
            self.get_logger().error("Service call error. %r" % (e,))
 
 
def main(args=None):
    rclpy.init(args=args)
    node = AddTwoIntsClient()
    rclpy.spin(node)
    rclpy.shutdown()
 
 
if __name__ == "__main__":
    main()